instrumentation/src/splunk.c

```diff
@@ -148,15 +155,26 @@ void set_env_var(logger log, const char *var_name, const char *value) {
 }
 
 void set_java_tool_options(logger log, struct config *cfg) {
-    char java_tool_options[ENV_VAR_LEN] = JAVA_TOOL_OPTIONS_PREFIX;
+    char java_tool_options[MAX_ENV_VAR_LEN] = JAVA_TOOL_OPTIONS_PREFIX;
     char log_line[MAX_LOG_LINE_LEN] = "";
     size_t jar_path_len = strlen(cfg->java_agent_jar);
     if (jar_path_len > MAX_CONFIG_ATTR_LEN) {
         sprintf(log_line, "jar_path too long: got %zu chars, max %d chars", jar_path_len, MAX_CONFIG_ATTR_LEN);
         log_warning(log, log_line);
         return;
     }
-    strcat(java_tool_options, (*cfg).java_agent_jar);
+    strncat(java_tool_options, cfg->java_agent_jar, MAX_ENV_VAR_LEN - strlen(java_tool_options) - 1);
```

Comment:
does it make sense to track remaining length for each of these updates instead of what may be silent noops with partial properties?

Reply:
I wasn't able to cause truncation using the maximum string lengths that the .so allows, but I've made some changes to check for truncation anyway to future proof this function (or in case I missed something), and I've added some unit tests to test the limits.